### DISCUSSION TITLE
Data driven line-dash-array styling

### DISCUSSION THREAD

**Discussion by wipfli** [maplibre/maplibre#167] [Cross-project feature proposals]
# Data driven line-dash-array styling

![image](https://user-images.githubusercontent.com/53421382/208367302-2493a83a-4c3a-496d-a267-fab028ecdb1d.png)

_image taken from http://osm.openmaptiles.org/#map=16/47.3900/8.4993&layers=V_

The lack of data-driven styling for line-dash-array is a significant headache for style designers, requiring additional layers if they want to use dashes. This discourages using dashes for technical reasons even if their cartography would be preferred.

See https://github.com/maplibre/maplibre-gl-js/issues/1235 for MapLibre GL JS issue

## Deliverables
1. Data-driven line-dash-array in MapLibre GL JS
2. Data-driven line-dash-array in MapLibre GL Native

Note: This could easily be split into two bounties, one for JS and one for Native.

Originally posted by @pnorman

**Comment by wipfli** [comment]
Thanks for this suggestion. I think it would be best to open an issue in MapLibre GL JS and Native if there is not already one for this issue.

**Reply by pnorman** [reply]
There is an issue in GL JS, linked above

**Comment by ovivoda** [comment]
Created an issue for Native as well here: [#744](https://github.com/maplibre/maplibre-gl-native/issues/744)
